src/main/java/de/tum/in/www1/jenkins/notifications/SendTestResultsNotificationPostBuildTask.java

```diff
@@ -173,11 +171,24 @@ private List<Commit> findCommits(Run<?, ?> run) {
             final Commit commit = new Commit();
             commit.setRepositorySlug(slug);
             commit.setHash(hash);
-
+            commit.setBranchName(getBranchName(buildData));
             return commit;
         }).collect(Collectors.toList());
     }
 
+    private @Nullable String getBranchName(BuildData buildData) {
+        String branchName = buildData.getLastBuiltRevision().getBranches().stream().map(GitObject::getName).findFirst().orElse(null);
+        if (branchName != null) {
```

Comment:
```suggestion
        if (branchName == null) {
            return null;
        }

        […]
```

Stylistic preference to avoid the indentation for the main logic currently inside the `if`-block, but can be left as-is as well.

Reply:
Thanks, I addressed it (and also prevented a NPE that could arise)